src/legacy/utils/streams/limit_stream.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Transform } from 'stream';
+
+export function createLimitStream(limit: number) {
+  let counter = 0;
+  return new Transform({
+    objectMode: true,
+    async transform(obj, enc, done) {
+      if (counter >= limit) {
+        return done(new Error(`Limit of ${limit} objects reached`));
```

Comment:
We should probably verify that the API behaves appropriately when more than 10000 objects are imported, since we're using `createConcatStream()` we don't have to worry about partial imports happening, but I'm pretty sure it will respond with a 500 and no meaningful error in this case, when ideally we should be responding with a 4xx response that instructs the user to send less objects. Maybe we want to be able to customize the behavior of `createLimitStream()` when the limit is encountered? Or maybe we should just define this Transform stream inline so we have full control of the behavior in import.ts?

Reply:
@spalger Interesting, I did the test and yeah it does throw a 500. Because of this, I moved `createLimitStream` inline to throw a 400 error. I've added functional tests for this as well.

As I changed this, I discovered the API doesn't accept files larger than 1MB. I had to add a config for what size we want to allow on import. I set the default to 50MB? The 10k object es archive is about 20MB when exported from the API.

Let me know your thoughts about these changes.

Reply:
Sounds good to me, but I wonder what Kibana performance is like when a 20 MB import is loaded into memory... maybe we should consider lowering the limit to 5000 or something

Reply:
As discussed and after doing some research, the import for 10k records takes about 3s and blocks the event loop up to ~800ms due to garbage collection. The fix will come when we use streams to not have all the objects in memory. We will leave the limit as is (limit is also configurable by users).